src/main/java/org/zalando/nakadi/domain/Timeline.java

```diff
@@ -0,0 +1,153 @@
+package org.zalando.nakadi.domain;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+public class Timeline {
```

Comment:
I don't understand your nullability policy here. When I look at this class it seems that everything can be null as you check everything for null. But then in repository you check for null only the latest position.